compiler/rustc_metadata/src/locator.rs

```diff
@@ -242,6 +243,7 @@ crate struct CrateLocator<'a> {
     // Immutable per-session configuration.
     sess: &'a Session,
     metadata_loader: &'a dyn MetadataLoader,
+    current_crate: Option<&'a str>,
```

Comment:
`current_crate` isn't used by `CrateLocator`, only by `CrateError::report`, so I'd pass it to that function instead (together with the rest of the `Session` as you wrote in https://github.com/rust-lang/rust/pull/88538#issuecomment-909456333).

Reply:
Reverted this commit.